scripts/build.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+const execa = require('execa');
+const program = require('commander');
+const path = require('path');
+const remove = require('lodash/remove');
+const EventEmitter = require('events');
+const { Worker } = require('worker_threads');
+
+const myEmitter = new EventEmitter();
+
+let modules;
+let i = 0;
+
+async function retrieveDependencies(_modules) {
+  const rtn = _modules;
+  const packages = await Promise.all(
+    rtn.map(pck => execa
+      .shell(
+        `lerna list -alp --include-filtered-dependencies --json --scope="${pck.name}"`,
+      )
+      .then(({ stdout }) => {
+        const dependents = JSON.parse(stdout).filter(
+          p => p.name !== pck.name,
+        );
+        return Object.assign(pck, { dependents });
+      })),
+  );
+  return packages;
+}
+
+function unstack() {
+  console.log(`-=-=-=- #${i} -=-=-=-`);
+  console.log(
+    `TODO (${modules.todo.length})`,
+    modules.todo.map(m => m.name).join(', '),
+  );
+  console.log(
+    `DOING (${modules.doing.length})`,
+    modules.doing.map(m => m.name).join(', '),
+  );
+  console.log(
+    `DONE (${modules.done.length})`,
+    modules.done.map(m => m.name).join(', '),
+  );
+  modules.todo.map((pck) => {
+    const deps = pck.dependents.filter(
+      dep => !modules.done.find(el => el.name === dep.name),
+    );
+
+    if (deps.length === 0) {
+      const workerData = remove(modules.todo, p => p.name === pck.name)[0];
+      modules.doing.push(workerData);
+
+      const promise = program.dryRun
+        ? new Promise(resolve => resolve())
+        : new Promise((resolve, reject) => {
+          const worker = new Worker(path.join(__dirname, '/build_module.js'), {
```

Comment:
Could the script be moved into a new directory (eg: scripts/worker)?

Because I find this quite confusing to have this script at the root of the `scripts` folder.

Reply:
Ok, if you prefer, but we should keep in mind that this script can be executed in standalone, without any worker ;)